compression/burrows_wheeler.py

```diff
@@ -0,0 +1,165 @@
+"""
+https://en.wikipedia.org/wiki/Burrows%E2%80%93Wheeler_transform
+
+The Burrows–Wheeler transform (BWT, also called block-sorting compression)
+rearranges a character string into runs of similar characters. This is useful
+for compression, since it tends to be easy to compress a string that has runs
+of repeated characters by techniques such as move-to-front transform and
+run-length encoding. More importantly, the transformation is reversible,
+without needing to store any additional data except the position of the first
+original character. The BWT is thus a "free" method of improving the efficiency
+of text compression algorithms, costing only some extra computation.
+"""
+
+
+def all_rotations(string):
+    """
+    :param str string: The string that will be rotated len(string) times.
+    :return: A list with len(string) rotations of the parameter string.
+    :rtype: list[str]
+    :raises TypeError: If the string parameter type is not str.
+    Examples:
+
+    >>> all_rotations("^BANANA|")
```

Comment:
`>>> all_rotations("^BANANA|")  # doctest: +NORMALIZE_WHITESPACE`

This allows you to lose the backslashes and indent the way your want because PEP8 says:
> The preferred way of wrapping long lines is by using Python’s implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.